### PR TITLE
Detail the reasons why PIP can't be triggered

### PR DESCRIPTION
**Pull request by beaufortfrancois**
FIX https://github.com/WICG/picture-in-picture/issues/22

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -122,16 +122,20 @@ and |options| is invoked, the user agent MUST run the following steps:
...
+    steps.
+3. Let |pipWindow| be the Picture In Picture window that is <dfn>always on
     top</dfn>. It consistently stays above most other windows.
-3. Let |pipVideo| be the video contained in the |pipWindow|.
-4. If a |pipWindow| is not already created, create one.
-5. If one can't be created, throw a {{UnknownError}} and abort these steps.
-6. Apply |options| {{width}}, {{height}}, {{PictureInPictureOptions/top}}, and
+4. Let |pipVideo| be the video contained in the |pipWindow|.
+5. If a |pipWindow| is not already created, create one.
+6. If one can't be created because user agent blocked feature for instance,
```
Comment (mounirlamouri) [inline index.bs]: What do you mean by "because of user agent blocked feature"?

File: index.bs [hunk 2]
```diff
@@ -122,16 +122,20 @@ and |options| is invoked, the user agent MUST run the following steps:
...
+    throw a {{NotAllowedError}} and abort these steps.
+7. Apply |options| {{width}}, {{height}}, {{PictureInPictureOptions/top}}, and
```
Comment (mounirlamouri) [inline index.bs]: Maybe add "OPTIONALLY,", here?

File: index.bs [hunk 3]
```diff
@@ -122,16 +122,20 @@ and |options| is invoked, the user agent MUST run the following steps:
...
     {{start}} to the |pipWindow| dimensions and position. If some |options|
     can't be applied, it SHOULD apply latest ones that were successfully set if
```
Comment (mounirlamouri) [inline index.bs]: If we add "Optionnaly", maybe the SHOULD could be dropped?

File: index.bs (hunk 2 above)
Comment (beaufortfrancois) [reply index.bs]: Done.

File: index.bs (hunk 3 above)
Comment (beaufortfrancois) [reply index.bs]: Done.

File: index.bs (hunk 1 above)
Comment (beaufortfrancois) [reply index.bs]: I've removed this step eventually, as we already check for `pictureInPictureEnabled` and return a promise rejected with `NotSupportedError`. WDYT?

**Comment by beaufortfrancois** [comment]
@mounirlamouri I've addressed your feedback above.

File: index.bs [hunk 4]
```diff
@@ -122,15 +122,17 @@ and |options| is invoked, the user agent MUST run the following steps:
...
-5. If one can't be created, throw a {{UnknownError}} and abort these steps.
-6. Apply |options| {{width}}, {{height}}, {{PictureInPictureOptions/top}}, and
-    {{start}} to the |pipWindow| dimensions and position. If some |options|
-    can't be applied, it SHOULD apply latest ones that were successfully set if
-    any.
+4. Let |pipVideo| be the video contained in the |pipWindow|.
+5. If a |pipWindow| is not already created, create one.
+6. OPTIONALLY, apply |options| {{width}}, {{height}},
+    {{PictureInPictureOptions/top}}, and {{start}} to the |pipWindow| dimensions
+    and position. If some |options| can't be applied, it applies latest ones that
```
Comment (mounirlamouri) [inline index.bs]: s/it applies/applies/ ?

Comment (beaufortfrancois) [reply index.bs]: Done.

**Comment by beaufortfrancois** [comment]
@mounirlamouri Fixed.

**Comment by mounirlamouri** [comment]
:+1: thanks :)

**Pull request merged**